pkg/reconciler/taskrun/taskrun_test.go

```diff
@@ -271,6 +272,29 @@ func getTaskRunController(t *testing.T, d test.Data) (test.Assets, func()) {
 	}, cancel
 }
 
+func checkEvents(fr *record.FakeRecorder, testName string, wantEvents []string) error {
+	timer := time.NewTimer(1 * time.Second)
+	foundEvents := []string{}
+	for ii := 0; ii < len(wantEvents)+1; ii++ {
```

Comment:
i find it a bit confusing that this function is both looping and receiving events via channel at the same time, but then after thinking through the alternative, this seems better - do you think it might be worth a comment explaining what is happening here and why?

Reply:
yeah, I can add one.
afaik the select case is the easiest way to multiplex N channels, so to handle the timeout which is needed in failure cases (fewer events then expected).